mlir/lib/ExecutionEngine/conv-validation-wrappers.cpp

```diff
@@ -844,41 +845,55 @@ mcpuConv2dInt8(int64_t rank1, void *f_ptr, int64_t rank2, void *i_ptr,
       dilation_h, dilation_w, xdlops);
 }
 
-int findIdxHistRelDiff(double relDiff) {
-  /*
-    hist_relDiff[]:
-    0: 0
-    1: 0 - 1e-6
-    2: 1e-6 - 1e-5
-    3: 1e-5 - 1e-4
-    4: 1e-4 - 1e-3
-    5: 1e-3 - 1e-2
-    6: 1e-2 - 0.1
-    7: 0.1 - 1
-    8: >= 1
-   */
+size_t findIdxHistRelDiff(double relDiff,
+                          const llvm::SmallVector<double, 7> bucketBoundary) {
```

Comment:
I don't think you want to pass vectors by value like this.
If anything, I'd go and make this a `static const double BUCKET_BOUNDARIES[NUM_BUCKETS] = {...};`
with a `constexpr size_t NUM_BUCKETS = 7;` before it

Reply:
But, failing that, take an `ArrayRef`